src/apple/macos/ffi.rs

```diff
@@ -38,89 +43,57 @@ extern "C" {
-        cStr: *const c_char,
-        encoding: CFStringEncoding,
-        contentsDeallocator: *mut c_void,
-    ) -> CFStringRef;
-
-    // Disk information functions are non-operational on iOS because of the sandboxing
-    // restrictions of apps, so they don't can't filesystem information. This results in
-    // mountedVolumeURLs and similar returning `nil`. Hence, they are MacOS specific here.
-
-    pub fn DASessionCreate(allocator: CFAllocatorRef) -> DASessionRef;
-
-    // pub fn DADiskCreateFromVolumePath(
-    //     allocator: CFAllocatorRef,
-    //     session: DASessionRef,
-    //     path: CFURLRef,
-    // ) -> DADiskRef;
-    pub fn DADiskCreateFromBSDName(
+    pub fn IORegistryEntryCreateCFProperty(
+        entry: io_registry_entry_t,
+        key: CFStringRef,
         allocator: CFAllocatorRef,
-        session: DASessionRef,
-        path: *const c_char,
-    ) -> DADiskRef;
-    // pub fn DADiskGetBSDName(disk: DADiskRef) -> *const c_char;
-
-    pub fn DADiskCopyDescription(disk: DADiskRef) -> CFMutableDictionaryRef;
-}
-
-pub type DADiskRef = *const __DADisk;
-pub type DASessionRef = *const __DASession;
-
-// We need to wrap `DASessionRef` to be sure `System` remains Send+Sync.
-pub struct SessionWrap(pub DASessionRef);
-
-impl Drop for SessionWrap {
-    fn drop(&mut self) {
-        if !self.0.is_null() {
-            unsafe {
-                CFRelease(self.0 as _);
-            }
-        }
-    }
+        options: IOOptionBits,
+    ) -> CFDictionaryRef;
+    pub fn IORegistryEntryGetParentEntry(
+        entry: io_registry_entry_t,
+        plane: io_name_t,
+        parent: *mut io_registry_entry_t,
+    ) -> libc::kern_return_t;
+
+    pub fn IOBSDNameMatching(
+        mainPort: mach_port_t,
+        options: u32,
+        bsdName: *const c_char,
+    ) -> CFMutableDictionaryRef;
+
+    // This is deprecated as of macOS 12.0, but Rust doesn't have a good way to only use the replacement on 12+.
+    pub static kIOMasterPortDefault: mach_port_t;
 }
 
-unsafe impl Send for SessionWrap {}
-unsafe impl Sync for SessionWrap {}
-
 #[cfg(all(
     not(feature = "apple-sandbox"),
     any(target_arch = "x86", target_arch = "x86_64")
 ))]
 mod io_service {
-    use super::mach_port_t;
+    use super::io_object_t;
+    use core_foundation_sys::base::mach_port_t;
+    use libc::{c_char, c_void, size_t};
 
-    #[allow(non_camel_case_types)]
-    pub type io_object_t = mach_port_t;
     #[allow(non_camel_case_types)]
     pub type io_connect_t = io_object_t;
-    #[allow(non_camel_case_types)]
-    pub type io_iterator_t = io_object_t;
+
+    extern "C" {
+        pub fn IOServiceMatching(a: *const c_char) -> *mut c_void;
+
+        pub fn IOServiceOpen(device: io_object_t, a: u32, t: u32, x: *mut io_connect_t) -> i32;
+
+        pub fn IOServiceClose(a: io_connect_t) -> i32;
+
+        #[allow(dead_code)]
+        pub fn IOConnectCallStructMethod(
+            connection: mach_port_t,
+            selector: u32,
```

Comment:
This is wrong here. From the [documentation](https://developer.apple.com/documentation/swift/int):

> On 32-bit platforms, Int is the same size as Int32, and on 64-bit platforms, Int is the same size as Int64.

Better make a type alias for it.

Reply:
Sorry, wrong line, it's for `inputStructCnt: size_t,` just below.

Reply:
I'm not sure that I follow here, sorry. [The docs](https://developer.apple.com/documentation/iokit/1514274-ioconnectcallstructmethod?language=objc) for `IOConnectCallStructMethod` say that `inputStructCnt` is a `size_t` in the header. The current definition here in `sysinfo` is, as you noted, `inputStructCnt: size_t`, where `size_t` is the `libc` import. Isn't that already correct?

Reply:
Sorry, I was looking at the wrong docs once again... This one to be exact: https://developer.apple.com/documentation/iokit/1514274-ioconnectcallstructmethod